lib/CL/devices/cuda/pocl-cuda.c

```diff
@@ -581,13 +542,53 @@ pocl_cuda_init (unsigned j, cl_device_id dev, const char *parameters)
 
   dev->svm_allocation_priority = 2;
   dev->svm_caps = CL_DEVICE_SVM_COARSE_GRAIN_BUFFER;
+  if (data->supports_managed_memory) {
+    dev->svm_allocation_priority = 2;
+
+    /* OpenCL 2.0 properties */
+    dev->svm_caps = CL_DEVICE_SVM_COARSE_GRAIN_BUFFER
+                    | CL_DEVICE_SVM_FINE_GRAIN_BUFFER;
+//                    | CL_DEVICE_SVM_ATOMICS;
+  }
+
+  /* OpenCL 3.0 properties */
+  /* Minimum mandated capability */
+  dev->atomic_memory_capabilities
+      = CL_DEVICE_ATOMIC_ORDER_RELAXED | CL_DEVICE_ATOMIC_SCOPE_WORK_GROUP;
+  dev->atomic_fence_capabilities = CL_DEVICE_ATOMIC_ORDER_RELAXED
+                                   | CL_DEVICE_ATOMIC_ORDER_ACQ_REL
+                                   | CL_DEVICE_ATOMIC_SCOPE_WORK_GROUP;
+
+  // TODO should check from Compute Capability ??
```

Comment:
Yep. I think safest to not assume this by default.

Reply:
fixed